shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/schedule/ExecutionStatusMonitor.java

```diff
@@ -177,19 +177,34 @@ private void decompressTGZ(File in, File out, CarminDatasetProcessing carminData
         try (TarArchiveInputStream fin = new TarArchiveInputStream(
                 new GzipCompressorInputStream(new FileInputStream(in)))) {
             TarArchiveEntry entry;
+
+            File cacheFolder = new File(out.getAbsolutePath() + File.separator + "cache");
```

Comment:
Dear Alae, thank you for your PR. When multiple threads run into that method, with the same result file name, what can possibly happen, they will write into the same cacheFolder. I would propose a cache folder either with an UUID or a timestamp with milliseconds to avoid overriding of different results. Thank you and with kind regards, Michael

Reply:
see microservice import, that generates a random ID for that purpose

Reply:
Hello @michaelkain,

We do actually use a timestamp for the results folder, we create the path in the front, and take it back from the carminDatasetProcessing entity resultsLocation line 118

![image](https://user-images.githubusercontent.com/44404385/203293402-beb0800d-fb19-48f1-b50f-89e8d2c5d585.png)

[https://github.com/alaeessaki/shanoir-ng/blob/adding-multiple-result-extension-support/shanoir-ng-front/src/app/processing/execution/execution.component.ts](url)
![image](https://user-images.githubusercontent.com/44404385/203293932-10c90589-7d91-4587-ae7c-ccb26e891477.png)

to recapitulate : 

- The user launch an execution
- A carminDatasetProcessing is created with a unique result path and saved in the database
- The thread checks if the execution is finished 
- Once the execution is finished we get carminDatasetProcessing with the unique  result path created previously
- Creation of the unique folder and the cache folder inside it 

I hope I explained the workflow, 

With kind regards.

Khalil

Reply:
hello @michaelkain, 
Hope you're doing well, 
We have already discussed the code in this PR; what we changed is only the way of importing files from a tar that contains folders, and we added multiple tar file extension support.
Thank you.

Reply:
Thank you for your answers!